workflow-templates/update_rubocop.yml

```diff
@@ -0,0 +1,57 @@
+name: Update Rubocop
+
+# This action automatically updates the rubocop gems within a
+# repository and attempt to ensure that the resulting build passes.
+# In order to use it effectively on a given project you should ensure:
+# * All Rubocop associated gems are in the 'lint' group in the Gemfile
+# * You update the 'BUNDLE_WITHOUT' section to exclude other groups, especially
+#   those which will block `bundle install`
+#
+# Once added this will add an actions which:
+# * Automatically runs at 10 am Sunday
+# * Can be triggered manually via a workflow_dispatch (See link below)
+# * Will automatically update all gems in the 'lint' group
+# * Will run new safe autocorrect cops, and will add inline rubocop:todo for
+#   other cops
+# * Will automatically generate a pull request with the above changes.
+#
+on:
+  # Can be triggered manually from the GH interface or via a dispatches POST
+  # https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow
+  workflow_dispatch:
+  schedule:
+  # * is a special character in YAML so you have to quote this string
+  # Every sunday at 10 am
+  - cron: "0 10 * * 0"
+
+jobs:
+  update_rubocop:
+    if: (github.repository_owner == 'sanger') # On the sanger fork only
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Ruby
+      uses: ruby/setup-ruby@v1
+      env:
+        BUNDLE_WITHOUT: 'test development default'
+      with:
+        bundler-cache: true
+    - name: Update Rubocop
+      run: |
+        bundle config --local deployment false
+        bundle update --group=lint
+    - name: Correcting and ignoring
+      run: bundle exec rubocop --auto-correct --disable-uncorrectable
+      # https://github.com/marketplace/actions/create-pull-request
+    - name: Create Pull Request
+      uses: peter-evans/create-pull-request@v3
+      with:
+        title: Automatic Rubocop update
+        branch: gh_action/update_rubocop
+        delete-branch: true
+        body: |
+          Automatic update via .github/workflows/update_rubocop.yml
```

Comment:
Full link might be useful here.

Reply:
It adds reasonable complexity to the action, you need something like this:
```
    - name: Generate url
        id: urls
        run: echo ::set-output name=action-url::"$GITHUB_SERVER_URL/$GITHUB_REPOSITORY/blob/$GITHUB_SHA/"
      # https://github.com/marketplace/actions/create-pull-request
    - name: Create Pull Request
      uses: peter-evans/create-pull-request@v3
      with:
        title: Automatic Rubocop update
        branch: gh_action/update_rubocop
        delete-branch: true
        body: Automatic update via ${{ steps.urls.outputs.action-url }}/.github/workflows/update_rubocop.yml
```